### ISSUE TITLE
Start-ServiceFabricClusterConfigurationUpgrade doesn't start a cluster configuration upgrade.

### ISSUE THREAD

**Issue by JustinKaffenberger** [Azure/service-fabric-issues#1572]
1. I export the configuration from my cluster using `Get-ServiceFabricClusterConfiguration`
2. I tweak the `ApplicationPorts` field to increase the number of ports available to Service Fabric, as a simple sample change to the config, and update the `ClusterConfigurationVersion` to `1.0.1`. The following is the resulting JSON that I store in `NewConfig.json`:

```json
{
    "Properties": {
        "GoalStateExpirationReminderInDays": 0,
        "Security": {
            "CertificateInformation": {
                "ClusterCertificate": {
                    "Thumbprint": "REDACTED",
                    "ThumbprintSecondary": "REDACTED",
                    "X509StoreName": "My"
                },
                "ServerCertificate": {
                    "Thumbprint": "REDACTED",
                    "ThumbprintSecondary": "REDACTED",
                    "X509StoreName": "My"
                },
                "ReverseProxyCertificate": {
                    "Thumbprint": "REDACTED",
                    "ThumbprintSecondary": "REDACTED",
                    "X509StoreName": "My"
                },
                "ClientCertificateThumbprints": [
                    {
                        "IsAdmin": true,
                        "CertificateThumbprint": "REDACTED"
                    }
                ],
                "ClientCertificateCommonNames": [
                    {
                        "IsAdmin": true,
                        "CertificateCommonName": "dev",
                        "CertificateIssuerThumbprint": "REDACTED"
                    }
                ]
            },
            "ClusterCredentialType": "X509",
            "ServerCredentialType": "X509"
        },
        "NodeTypes": [
            {
                "Name": "NodeType0",
                "ClientConnectionEndpointPort": 19000,
                "HttpGatewayEndpointPort": 19080,
                "ReverseProxyEndpointPort": 19081,
                "LeaseDriverEndpointPort": 19002,
                "ClusterConnectionEndpointPort": 19001,
                "ServiceConnectionEndpointPort": 19003,
                "ApplicationPorts": {
                    "StartPort": 20001,
                    "EndPort": 21024
                },
                "EphemeralPorts": {
                    "StartPort": 49152,
                    "EndPort": 65535
                },
                "IsPrimary": true
            }
        ],
        "AddonFeatures": [],
        "FabricSettings": [
            {
                "Name": "Setup",
                "Parameters": [
                    {
                        "Name": "FabricDataRoot",
                        "Value": "C:\\ProgramData\\SF"
                    },
                    {
                        "Name": "FabricLogRoot",
                        "Value": "C:\\ProgramData\\SF\\Log"
                    }
                ]
            }
        ],
        "EnableTelemetry": true,
        "DiagnosticsStore": {
            "StoreType": "FileShare",
            "DataDeletionAgeInDays": 21,
            "Connectionstring": "C:\\ProgramData\\SF\\DiagnosticsStore",
            "IsEncrypted": false
        },
        "FabricClusterAutoupgradeEnabled": true
    },
    "Nodes": [
        {
            "NodeName": "Node_4",
            "NodeTypeRef": "NodeType0",
            "IPAddress": "SFNODE5",
            "FaultDomain": "fd:/dc1/r4",
            "UpgradeDomain": "UD4"
        },
        {
            "NodeName": "Node_3",
            "NodeTypeRef": "NodeType0",
            "IPAddress": "SFNODE4",
            "FaultDomain": "fd:/dc1/r3",
            "UpgradeDomain": "UD3"
        },
        {
            "NodeName": "Node_2",
            "NodeTypeRef": "NodeType0",
            "IPAddress": "SFNODE3",
            "FaultDomain": "fd:/dc1/r2",
            "UpgradeDomain": "UD2"
        },
        {
            "NodeName": "Node_1",
            "NodeTypeRef": "NodeType0",
            "IPAddress": "SFNODE2",
            "FaultDomain": "fd:/dc1/r1",
            "UpgradeDomain": "UD1"
        },
        {
            "NodeName": "Node_0",
            "NodeTypeRef": "NodeType0",
            "IPAddress": "SFNODE1",
            "FaultDomain": "fd:/dc1/r0",
            "UpgradeDomain": "UD0"
        }
    ],
    "ApiVersion": "06-2018",
    "Name": "dev",
    "ClusterConfigurationVersion": "1.0.1"
}
```

3. I run `Start-ServiceFabricClusterConfigurationUpgrade -ClusterConfigPath .\NewConfig.json -Debug` to kick off the upgrade process. I receive absolutely no errors or feedback. I am connected to the cluster using the administrative certificate.
4. I run `Get-ServiceFabricClusterConfigurationUpgradeStatus` to see the status of the upgrade.
5. The status never changes, it remains as:
```
           UpgradeState ProgressStatus ConfigVersion Details
           ------------ -------------- ------------- -------
RollingForwardCompleted              0 1.0.0
```
6. I am unable to update cluster configuration, making it impossible to alter its state in production.

**Comment by JustinKaffenberger** [comment]
This issue of the 'no-op' occurring is only after I had to remove to the empty `WindowsIdentities` element from the JSON returned by Service Fabric, described in this [other issue](https://github.com/Azure/service-fabric-issues/issues/1512) that I've created. Overall the configuration system seems to have some serious issues with simple X509 secured clusters.

**Comment by BharatNarasimman** [comment]
+ @maburlik

**Comment by TonyChengCME** [comment]
I am also experiencing this behavior for a standalone five-node Service Fabric cluster running v7.0.457.9590, so this has not been resolved in the latest release.

**Comment by JustinKaffenberger** [comment]
After a bunch of time fiddling with different configurations, I've found the root cause of the problem to be the existence of duplicate thumbprints in the `Thumbprint` and `ThumbprintSecondary` JSON fields. In my naivety I assumed that was a valid configuration, at least until I needed to do a certificate rollover. 

By re-creating the cluster with only `Thumprint` populated I was able to once again upgrade the cluster configuration.
